### PR TITLE
Image related updates

### PR DESCRIPTION
**Pull request by ncdc**
Add information about the integrated Docker registry.

Update ImageStream docs.

### CODE REVIEW

**Comment by ncdc** [comment]
@smarterclayton PTAL

Some of the registry content seems a bit scattered (infra components->image registry vs admin guide->docker registry). I'm not sure what makes the most sense, organization-wise.

**Comment by ncdc** [comment]
cc @danmcp

**Comment by adellape** [comment]
@ncdc Could you rebase this when you get a chance?

Also, do you want to wait on more feedback or can I merge once it passes? I read through and didn't have any immediate comments/problems; planning to do a follow up PR to work on edits and organization per various [imageregistry]-tagged doc cards I have.

**Comment by ncdc** [comment]
@adellape will rebase shortly. I would like feedback from someone else in engineering if possible

@jwhonce @smarterclayton @danmcp

**Comment by adellape** [comment]
@jwhonce @smarterclayton @danmcp Looking for Eng feedback before merging.

File: admin_guide/docker_registry.adoc [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+- any deployment configs
+- any build configs
+- any builds
+- the 3 most recent items in stream.status.tags[].items (3 is configurable)
+
+There is no support for pruning from external registries.
+
+Images "managed by OpenShift" will have an annotation indicating this fact. The current proposed annotation is openshift.io/image.managed.
+
+The minimum age criterion should default to 60 minutes.
```
Comment (jwhonce) [inline admin_guide/docker_registry.adoc]: 'should' or 'does' default to 60 minutes?  When would the default not be 60 minutes?

File: admin_guide/docker_registry.adoc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+When removing an image, remove all references to the image from all ImageStreams having a reference to the image in `status.tags`.
```
Comment (jwhonce) [inline admin_guide/docker_registry.adoc]: Are you telling me to remove all reference or will the software remove all references when it removes an image?

File: admin_guide/docker_registry.adoc [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+- `--dry-run` - displays what images and layers would be pruned, but does not perform any real deletions. Defaults to `true`.
+- `--keep-younger-than` - only prune if the image is older than this value, and if the pods and/or image streams that reference the image are older than this value. Defaults to 60 minutes.
+- `--keep-tag-revisions` - only prune image revisions for a tag in an image stream's `status.tags` older than this value. Defaults to 3.
+
+The following is sample output from a dry run:
+
+[options="nowrap"]
+----
+$ osadm prune images --keep-younger-than=0 --keep-tag-revisions=1
```
Comment (jwhonce) [inline admin_guide/docker_registry.adoc]: Is the command 'prune-images' or 'prune images'?

File: admin_guide/docker_registry.adoc [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+When removing an image, remove all references to the image from all ImageStreams having a reference to the image in `status.tags`.
+
+Also automatically remove any image layer that is no longer referenced by any images.
+
+=== Running `osadm prune-images`
+
+The command to prune images and layers is `osadm prune-images`. It supports the following options:
+
+- `--dry-run` - displays what images and layers would be pruned, but does not perform any real deletions. Defaults to `true`.
```
Comment (jwhonce) [inline admin_guide/docker_registry.adoc]: Does imply I have to use --no-dry-run to actually have the command work?  Do other commands default to true for dry-run?

File: admin_guide/docker_registry.adoc (hunk 3 above)
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: `prune images`; fixed

File: admin_guide/docker_registry.adoc (hunk 4 above)
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: You have to use `--dry-run=false`, yes. The other prune subcommands (builds, deployments) default to true.

Comment (smarterclayton) [inline admin_guide/docker_registry.adoc]: Maybe we should just be consistent.

----- Original Message -----

File: admin_guide/docker_registry.adoc (hunk 1 above)
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: fixed

File: admin_guide/docker_registry.adoc (hunk 2 above)
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: fixed

File: admin_guide/docker_registry.adoc (hunk 4 above)
Comment (ncdc) [inline admin_guide/docker_registry.adoc]: I'm pretty sure we are

**Comment by tnguyen-rh** [comment]
@ncdc (re --dry-run)
I think being self-consistent is important.  However, IMHO, being consistent
w/ the world at large is better, in this case.  The world at large usually
implements --dry-run as a simple flag (no argument), whose presence renders the
command non-effective (but w/ "would do" messages), and whose absence allows
the command to proceed normally (effective).  Departing from this adds a huge
documentation burden, and risks confusing the users, for no real benefit.
Please reconsider.

**Comment by ncdc** [comment]
@tnguyen-rh I can't say that I see this as a "huge" doc burden, but if everyone agrees, I'd be fine renaming `--dry-run` to something like `--force` or `--confirm` and requiring that the user specify the flag to make pruning happen. Note, this also applies to pruning builds and deployments (see #391).

@smarterclayton @jwhonce @danmcp @derekwaynecarr WDYT?

**Comment by adellape** [comment]
+1 on `--confirm`.

**Comment by smarterclayton** [comment]
I'm fine with --confirm as our "do the one way thing but not by default".  Force is reserved for doing something dangerous that isn't always correct, so we want can't use that.

**Comment by jwhonce** [comment]
+1 on --confirm

**Comment by adellape** [comment]
Checked with Andy: going to merge this as-is now and update later after we switch to `--confirm`.

**Pull request merged**